### PR TITLE
Added and used the requestHandler function issue #4

### PR DESCRIPTION
**Pull request by devanshkejriwal**
This will close issue #4. This is what I've done till now. I'm hoping to test it in some way.

### CODE REVIEW

**Comment by Jake-Convictional** [comment]
To test this, copy the `api.js` file from convictional-node repository and manually overwrite the `api.js` file in the front end repository's node_modules folder.

File: lib/functions.js [hunk 1]
```diff
@@ -13,3 +13,21 @@ module.exports = {
...
+          } else {
+              reject(new Error('A problem occurred with the request. Code: ' + res.statusCode))
+          }
+      } else {
+          happyPath(res, body)
+      }
+  }
+}
+
+exports.requestHandler = requestHandler
```
Comment (Jake-Convictional) [inline lib/functions.js]: Nit: newline at EOF

**Comment by devanshkejriwal** [comment]
> This looks good, make sure you get a chance to test different scenarios (200, 201, 404, 500, connection error)

How do I do this? I finished copying the files

**Comment by Jake-Convictional** [comment]
I believe what I did before was to edit the routes in api.js to point to pages that I knew would return 404, 500 etc. You can use https://httpstat.us/ for this. Let me know if you run into any problems

**Pull request merged**